services/navigation/builtin/builtin.go

```diff
@@ -93,16 +98,26 @@ func (conf *Config) Validate(path string) ([]string, error) {
 	}
 	deps = append(deps, conf.BaseName)
 
-	if conf.MovementSensorName == "" {
-		return nil, utils.NewConfigValidationFieldRequiredError(path, "movement_sensor")
-	}
-	deps = append(deps, conf.MovementSensorName)
-
 	if conf.MotionServiceName == "" {
 		conf.MotionServiceName = resource.DefaultServiceName
 	}
 	deps = append(deps, resource.NewName(motion.API, conf.MotionServiceName).String())
 
+	if conf.MapType != "" {
+		if !slices.Contains([]string{"None", "GPS"}, conf.MapType) {
+			return nil, errors.New("invalid map_type, when defined map_type must be one of the following ['None', 'GPS']")
+		}
+	} else {
+		conf.MapType = defaultMapType
```

Comment:
I thought that we can't change anything in the config, since that will trigger a reconfig (thinking of https://viam.atlassian.net/browse/RSDK-2252).

Reply:
That is a great question and one I had as well. Based on the other variables being set to default values in Validate (https://github.com/viamrobotics/rdk/blob/ed7df66139c1cb3d3d6a1d6cdf4b0f08e57990c4/services/navigation/builtin/builtin.go#L114), I thought I may have misunderstood how/when reconfigure gets triggered.

Reply:
Yeah, I'm confused. What we saw in that ticket was that modifying fields in the config caused rdk to detect that the config was out of sync from the cloud config and trigger a reconfig. I suggest either asking the SDK team to take a look or testing this manually by running viam-server (connected to app) with a navigation service where the map type gets filled in from the default and seeing if it triggers a reconfig.

Reply:
Investigating now, will update this thread with results

Reply:
After some investigation, it appears that reconfigure and validate are working as intended. When the map_type is changed reconfigure only occurs once. I spoke with @benjirewis on this matter and we figure that either:
- `Validate` is called on the config coming from the cloud before it is compared to the old config stored in the resource graph
OR
- The config stored in the resource graph does not contain the defaults set during `Validate` and so the comparison passes

Benji will be doing some additional investigation on this end. He has also informed me that this is not a pattern we truly want to support (despite it working presently) and he would advise moving the setting of defaults into `Reconfigure`. @tessavitabile would you prefer I handle this change over now or leave it as is and just add a comment linked to a JIRA ticket to be triaged and done at a future time?

Reply:
Thank you for the investigation! I think it's better to make this change now, while you're working in the code.

Reply:
> `Validate` is called on the config coming from the cloud before it is compared to the old config stored in the resource graph

Confirmed locally; this is what's happening. `Reconfigure` knows to only reconfigure the robot when there is a difference between the config processed from the cloud and the config generated from the resource graph. Processing the config from the cloud involves multiple calls to your `Validate` method, which mutate the `ConvertedAttributes` field of your navigation config. That mutation exists before `Reconfigure` even starts. So, the config is stored in the resource graph with that mutated `ConvertedAttributes` field (defaults applied). When we pull a cloud config again, `Validate` mutates the `ConvertedAttributes` field again, and when comparing to the resource graph, no salient diff is found.

What this means is that the current design _works_, but not in a way I or the SDKs team were aware was supported. Seems like you're already heading down this path, but I'd suggest applying default values in the constructor/`Reconfigure` and not in `Validate`.

Separately, I wish the signature for `Validate` methods did not pass the config by pointer so that no one would be tempted to mutate the config. I'll make a PR at some point to warn users against mutating the config passed to `Validate`.